devices/metal/metal_engine.h

```diff
@@ -6,6 +6,12 @@
 
 #include "core/engine.h"
 #include "metal_device.h"
+#include "TargetConditionals.h"
```

Comment:
Should be included as `<TargetConditionals.h>`. I also think that `platform.h` would be a better place for this.

Reply:
addressed in recent commits